ml-proto/src/spec/arithmetic.ml

```diff
@@ -119,10 +123,30 @@ struct
 
   let unop op =
     let f = match op with
-      | Clz -> fun i -> i  (* TODO *)
-      | Ctz -> fun i -> i  (* TODO *)
-      | Popcnt -> fun i -> i  (* TODO *)
-    in fun v -> Int.to_value (f (Int.of_value 1 v))
+      | Clz -> let rec loop acc n =
+                 match n with 
+                     | n when n = Int.zero -> Int.size 
+                     | n when Int.logand n (Int.shift_left Int.one (Int.size-1))
+                              <> Int.zero
+                         -> acc 
+                     | _ -> loop (1+acc) (Int.shift_left n 1) 
+               in loop 0
+      | Ctz -> let rec loop acc n =
+                 match n with 
+                     | n when n = Int.zero -> Int.size
+                     | n when Int.logand n Int.one = Int.one -> acc
+                     | _ -> loop (1+acc) (Int.shift_right_logical n 1) 
+               in loop 0
+      | Popcnt -> let rec loop acc i n =
+                    if n = Int.zero then acc
+                    else let acc = 
+                           if Int.logand n Int.one = Int.one
```

Comment:
Nit: odd indentation. Also, the shadowing of `acc` is confusing. I'd write this as
```
let open Int in
let rec loop acc i n =
  if n = zero then
    acc
  else
    let acc' = if logand n one = one then acc + 1 else acc in
    loop acc' (i - 1) (shift_right_logical n 1)
```